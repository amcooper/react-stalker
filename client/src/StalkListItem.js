import React from "react";
import { Link } from "react-router-dom";

function StalkListItem(props) {
  return (
    <Link to={`/${props.item.id}`}>
      <div className="StalkListItem" onClick={() => props.onClick()}>
        <p>
          {props.item.celebrity} on {props.item.date.toDateString()}
        </p>
      </div>
    </Link>
  );
}

export default StalkListItem;
