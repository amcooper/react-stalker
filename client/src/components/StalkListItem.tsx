import React from "react";
import { Link } from "react-router-dom";

function StalkListItem(props) {
  return (
    <Link to={`/${props.item.id}`}>
      <p className="StalkListItem" onClick={() => props.onClick()}>
        {props.item.celebrity} on {props.item.date.toDateString()}
      </p>
    </Link>
  );
}

export default StalkListItem;
