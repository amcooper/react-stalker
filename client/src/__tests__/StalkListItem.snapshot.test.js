import React from "react";
import StalkListItem from "../components/StalkListItem";
import { MemoryRouter } from "react-router-dom";
import { create } from "react-test-renderer";
import { stalkList } from "../fixtures/fixtures";

const testStalk = stalkList[0];

describe("StalkListItem component renders a stalk list item correctly", () => {
  it("renders correctly", () => {
    const rendered = create(
      <MemoryRouter>
        <StalkListItem item={testStalk} />
      </MemoryRouter>
    );

    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
