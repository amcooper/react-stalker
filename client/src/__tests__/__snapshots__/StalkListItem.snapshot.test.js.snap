// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StalkListItem component renders a stalk list item correctly renders correctly 1`] = `
<a
  href="/1"
  onClick={[Function]}
>
  <p
    className="StalkListItem"
    onClick={[Function]}
  >
    Rosie Perez
     on 
    Mon Apr 23 2018
  </p>
</a>
`;
